mod/state-transition/pkg/core/state_processor_withdrawals.go

```
@@ -0,0 +1,267 @@
+	}
+
+	totalValidators, err := st.GetTotalValidators()
+	if err != nil {
+		return err
+	}
+
+	// Update the next validator index to start the next withdrawal sweep.
+	var nextValidatorIndex math.ValidatorIndex
+
+	//#nosec:G701 // won't overflow in practice.
+	if len(expectedWithdrawals) == int(sp.cs.MaxWithdrawalsPerPayload()) {
```

golangci-lint (1.62.2)

[high] G115: integer overflow conversion uint64 -> int

(gosec)